cwms-data-api/src/main/java/cwms/cda/data/dto/Property.java

```diff
@@ -0,0 +1,149 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.data.dto;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV2;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.util.Objects;
+
+@XmlRootElement(name = "property")
+@XmlAccessorType(XmlAccessType.FIELD)
+@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class)
+@JsonDeserialize(builder = Property.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public final class Property implements CwmsDTOBase {
+
+    private final String category;
+    private final String name;
+    private final String office;
+    private final String value;
+
+    private Property(Builder builder) {
+        this.category = builder.category;
+        this.name = builder.name;
+        this.office = builder.office;
+        this.value = builder.value;
+    }
+    
+    @Override
+    public void validate() throws FieldException {
+
+        if (this.category == null || this.category.trim().isEmpty()) {
+            throw new FieldException("The 'category' field of a Property cannot be null or empty.");
+        }
+        if (this.name == null || this.name.trim().isEmpty()) {
+            throw new FieldException("The 'name' field of a Property cannot be null or empty.");
+        }
+        if (this.office == null || this.office.trim().isEmpty()) {
+            throw new FieldException("The 'office' field of a Property cannot be null or empty.");
+        }
+        if (this.value == null || this.value.trim().isEmpty()) {
+            throw new FieldException("The 'value' field of a Property cannot be null or empty.");
```

Comment:
This seems... odd?

Reply:
Which part?  That the validate does anything at all or that value can't be null/empty?

Reply:
I just checked, prop_value and prop_comment are both Nullable in the database.

Reply:
good reminder that I forgot the comment. My understanding is that this method is for validating client submissions, what is the case where we'd accept a null property value? If we want to delete, we'd use the delete function.

Reply:
Hmm, fair point, but "Property EXISTS and is set to NULL" does equal "Property does not exist". 
One is far more extreme than the other.

Reply:
ok, updated to remove validation on value